### PR TITLE
fix: add missing try-catch to services

### PR DESCRIPTION
**Pull request by RonenMars**
## What type of PR is this? (check all applicable)

- [ ] 💡 (feat) - A new feature (non-breaking change which adds functionality)
- [ ] 🔄 (refactor) - Code Refactoring - A code change that neither fixes a bug nor adds a feature
- [x] 🐞 (fix) - Bug Fix (non-breaking change which fixes an issue)
- [ ] 🏎 (perf) - Optimization
- [ ] 📄 (docs) - Documentation - Documentation only changes
- [ ] 📄 (test) - Tests - Adding missing tests or correcting existing tests
- [ ] 🎨 (style) - Styles - Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
- [ ]  ⚙️ (ci) - Continuous Integrations - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
- [ ]  ☑️ (chore) - Chores - Other changes that don't modify src or test files
- [ ]  ↩️ (revert) - Reverts - Reverts a previous commit(s).

## Description
Add try-catch blocks in the missing calls in services.

### CODE REVIEW

File: src/services/deployments.service.ts [hunk 1]
```diff
@@ -43,25 +43,30 @@ export class DeploymentsService {
...
+			}
 
-		const environmentIds = getIds(environments!, "envId");
-		const { data: projectDeployments, error: deploymentsError } = await this.listByEnvironmentIds(environmentIds);
+			const environmentIds = getIds(environments!, "envId");
+			const { data: projectDeployments, error: deploymentsError } = await this.listByEnvironmentIds(environmentIds);
 
-		if (deploymentsError) {
-			LoggerService.error(namespaces.deploymentsService, (deploymentsError as Error).message);
+			if (deploymentsError) {
```
Comment (efiShtain) [inline src/services/deployments.service.ts:58]: deployment error? are you trying to deploy something?

File: src/services/projects.service.ts [hunk 2]
```diff
@@ -34,50 +34,60 @@ export class ProjectsService {
 	}
 
 	static async build(projectId: string, resources: Record<string, Uint8Array>): Promise<ServiceResponse<string>> {
-		const { error: resourcesError } = await this.setResources(projectId, resources);
-		if (resourcesError) {
-			LoggerService.error(`${namespaces.projectService} - Upload resources`, (resourcesError as Error).message);
+		try {
+			const { error: resourcesError } = await this.setResources(projectId, resources);
+			if (resourcesError) {
+				LoggerService.error(`${namespaces.projectService} - Upload resources`, (resourcesError as Error).message);
```
Comment (efiShtain) [inline src/services/projects.service.ts:40]: missing context in error, you need the project id here

Comment (efiShtain) [inline src/services/projects.service.ts:40]: this does the same thing as this.setResources
why do you need both ?

File: src/services/projects.service.ts [hunk 3]
```diff
@@ -34,50 +34,60 @@ export class ProjectsService {
...
 		}
-		return { data: buildId, error: undefined };
 	}
 
 	static async deploy(projectId: string, buildId: string): Promise<ServiceResponse<string>> {
-		const { data: environments, error: envError } = await EnvironmentsService.listByProjectId(projectId);
-		if (envError) {
-			return { data: undefined, error: envError };
-		}
+		try {
```
Comment (efiShtain) [inline src/services/projects.service.ts:62]: is this try catch has any meaning ?
you wrapped all your low level services with try catch and return {data, error} from them
why do you wrap it here again with try catch ?
what can happen ?

File: src/services/deployments.service.ts (hunk 1 above)
Comment (RonenMars) [reply src/services/deployments.service.ts:58]: refactored

File: src/services/projects.service.ts (hunk 2 above)
Comment (RonenMars) [reply src/services/projects.service.ts:40]: removed

File: src/services/projects.service.ts (hunk 3 above)
Comment (RonenMars) [reply src/services/projects.service.ts:62]: If we sending invalid data on `projectsClient.build`, we can get an unhandled exception without any awareness from the user side. Something that happened when we sent invalid data on Windows.

File: src/services/deployments.service.ts [hunk 4]
```diff
@@ -43,25 +43,31 @@ export class DeploymentsService {
...
 
-		const environmentIds = getIds(environments!, "envId");
-		const { data: projectDeployments, error: deploymentsError } = await this.listByEnvironmentIds(environmentIds);
+			const environmentIds = getIds(environments!, "envId");
+			const { data: projectDeployments, error: listDeploymentsByEnvIdsError } =
+				await this.listByEnvironmentIds(environmentIds);
 
-		if (deploymentsError) {
-			LoggerService.error(namespaces.deploymentsService, (deploymentsError as Error).message);
+			if (listDeploymentsByEnvIdsError) {
```
Comment (RonenMars) [inline src/services/deployments.service.ts:59]: `listDeploymentsErrr]or`

File: src/services/projects.service.ts [hunk 5]
```diff
@@ -34,50 +34,58 @@ export class ProjectsService {
...
 	static async build(projectId: string, resources: Record<string, Uint8Array>): Promise<ServiceResponse<string>> {
-		const { error: resourcesError } = await this.setResources(projectId, resources);
-		if (resourcesError) {
-			LoggerService.error(`${namespaces.projectService} - Upload resources`, (resourcesError as Error).message);
-
-			return { data: undefined, error: resourcesError };
-		}
+		try {
+			const { error: resourcesError } = await this.setResources(projectId, resources);
+			if (resourcesError) {
```
Comment (RonenMars) [inline src/services/projects.service.ts:39]: Need log, check

File: src/services/projects.service.ts (hunk 3 above)
Comment (RonenMars) [reply src/services/projects.service.ts:60]: to remove

**Pull request merged**